### PR TITLE
Document Map stdlib

### LINKED ISSUE #74: Document Map stdlib
https://github.com/grain-lang/grain/blob/master/stdlib/map.gr

### PR DESCRIPTION
**Pull request by ospencer**
Closes #74

### CODE REVIEW

File: src/stdlib/map.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+title: Map
+---
+
+A Map holds key-value pairs. Any value may be used as a key or value.
```
Comment (phated) [inline src/stdlib/map.md:5]:
```suggestion
A Map holds key-value pairs. Any value may be used as a key or value. Operations on a `Map` mutate the internal state, so it never needs to be re-assigned.
```

File: src/stdlib/map.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+Retrieves the value associated with the given key from the map.
+
+### Map.**set**
+
+```grain
+set : (k, v, Map<k, v>) -> Void
+```
+
+Adds a new key-value pair to the map.
```
Comment (phated) [inline src/stdlib/map.md:43]: I forget, does it replace the value if it is already there?

File: src/stdlib/map.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Creates a new, empty map with an initial storage size for the given number of elements.
+
+### Map.**get**
+
+```grain
+get : (k, Map<k, v>) -> v
+```
+
+Retrieves the value associated with the given key from the map.
```
Comment (phated) [inline src/stdlib/map.md:35]: Fails if it is not there?

**Review by phated — CHANGES_REQUESTED** [review]

File: src/stdlib/map.md (hunk 3 above)
Comment (ospencer) [reply src/stdlib/map.md:35]:
```suggestion
Retrieves the value associated with the given key from the map. Fails if the key does not exist.
```

File: src/stdlib/map.md (hunk 2 above)
Comment (ospencer) [reply src/stdlib/map.md:43]: Yeah, it replaces.

Comment (ospencer) [reply src/stdlib/map.md:43]:
```suggestion
Adds a new key-value pair to the map. If the key already exists, replaces the value.
```

**Review by phated — APPROVED** [review]
💥

**Pull request merged**